libs/repositories/src/UsdRepository/UsdRepositoryCoingecko.spec.ts

```diff
@@ -0,0 +1,132 @@
+import { Container } from 'inversify';
+import { UsdRepositoryCoingecko } from './UsdRepositoryCoingecko';
+import { SupportedChainId } from '../types';
+import { assert } from 'console';
+
+const FIVE_MINUTES = 5 * 60 * 1000;
+const ONE_HOUR = 60 * 60 * 1000;
+const ONE_DAY = 24 * 60 * 60 * 1000;
+const WETH = '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2';
+const DEFINITELY_NOT_A_TOKEN = '0x0000000000000000000000000000000000000000';
+
+describe('UsdRepositoryCoingecko', () => {
+  let usdRepositoryCoingecko: UsdRepositoryCoingecko;
+
+  beforeAll(() => {
+    const container = new Container();
+    container
+      .bind<UsdRepositoryCoingecko>(UsdRepositoryCoingecko)
+      .to(UsdRepositoryCoingecko);
+    usdRepositoryCoingecko = container.get(UsdRepositoryCoingecko);
+  });
+
+  describe('getUsdPrice', () => {
+    it('should return the current price of WETH', async () => {
+      const price = await usdRepositoryCoingecko.getUsdPrice(
+        SupportedChainId.MAINNET,
+        WETH
+      );
+
+      expect(price).toBeGreaterThan(0);
```

Comment:
No, its not a mock. It was very useful to do the real call. I wanted also to ensure the result is aligned with my expectations (the number of points and distant among them).

Thechnically I think is not a e2e, its an integration test. I renamed the file to `.test.ts` instead of `.spec.ts`, so we can use `.test.ts` for integration test. Would that suffice?

Reply:
Right, integration rather than e2e.
Still `spec` vs `test` means the same IMO == unit tests.
Would be possible to have this somewhere else making it explicit it's an integration test?